Documentation/BuildInstructionsMacOS.md

```diff
@@ -0,0 +1,47 @@
+# macOS prerequisites
+
+Make sure you have all the dependencies installed:
+
+```console
+# core
+brew install coreutils qemu bash gcc@10 ninja cmake ccache rsync
+
+# (option 1) fuse + ext2
+brew install e2fsprogs m4 autoconf automake libtool
+brew install --cask osxfuse
+Toolchain/BuildFuseExt2.sh
+
+# (option 2) genext2fs
+brew install genext2fs
+```
+
+Notes:
+
+- fuse-ext2 is not available as brew formula so it must be installed using `BuildFuseExt2.sh`
+- Xcode and `xcode-tools` must be installed (`git` is required by some scripts)
+- coreutils is needed to build gcc cross compiler
+- qemu is needed to run the compiled OS image. You can also build it using the `BuildQemu.sh` script
+- osxfuse, e2fsprogs, m4, autoconf, automake, libtool and `BuildFuseExt2.sh` are needed if you want to build the root filesystem disk image natively on macOS. This allows mounting an EXT2 fs and also installs commands like `mke2fs` that are not available on stock macOS.
+- Installing osxfuse for the first time requires enabling its system extension in System Preferences and then restarting your machine. The output from installing osxfuse with brew says this, but it's easy to miss.
+- bash is needed because the default version installed on macOS doesn't support globstar
+- If you install some commercial EXT2 macOS fs handler instead of osxfuse and fuse-ext2, you will need to `brew install e2fsprogs` to obtain `mke2fs` anyway.
+- As of 2020-08-06, you might need to tell the build system about your newer host compiler. Once you've built the toolchain, navigate to `Build/i686/`, `rm -rf *`, then run `cmake ../.. -G Ninja -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10`, then continue with `ninja install` as usual.
```

Comment:
I think this one should be much higher and listed as a strict requirement, people constantly need to add this after posting in build problems.

Reply:
Aye, that's a very good idea. Also, upon closer reading most of those notes are not actually helping anything useful for someone who's building the system for the first time.

Reply:
I've removed this bit entirely and just updated `Meta/serenity.sh` to auto-detect this instead.